File: codegen/results/v001/qwen3.5_4b-nvfp4_think/20260409_092646/typescript/001_csv_to_json/solution.ts
import { readFileSync } from 'fs';

const content = readFileSync('./input/data.csv', 'utf-8');

const lines = content.split('\n').filter(line => line.trim());

if (lines.length === 0) {
  process.stdout.write('[]\n');
  process.exit(0);
}

const headerLine = lines[0].split(',');
const columnNameToIndex: Map<string, number> = new Map();

for (let i = 0; i < headerLine.length; i++) {
  const colName = headerLine[i].trim().toLowerCase();
  if (colName === 'name') columnNameToIndex.set('name', i);
  else if (colName === 'age') columnNameToIndex.set('age', i);
  else if (colName === 'email') columnNameToIndex.set('email', i);
  else if (colName === 'score') columnNameToIndex.set('score', i);
}

const data: Array<{ Name: string; Age: number; Email: string; Score: number }> = [];

for (let i = 1; i < lines.length; i++) {
  const columns = lines[i].split(',');
  
  let entry: any = {};
  for (const [name, index] of columnNameToIndex.entries()) {
    if (index >= 0 && index < columns.length) {
      value = columns[index];
      if (name === 'age') entry['Age'] = parseInt(value.trim(), 10);
      else if (name === 'score') entry['Score'] = parseFloat(value.trim());
      else if (name === 'name' || name === 'email') entry[name] = value.trim();
    }
  }
  
  data.push(entry as any);
}

console.log(JSON.stringify(data));

File: codegen/results/v001/qwen3.5_4b-nvfp4_think/20260409_092646/go/001_csv_to_json/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"os"
)

type Person struct {
	Name   string
	Age    int
	Email  string
	Score  float64
}

func main() {
	file, err := os.Open("input/data.csv")
	if err != nil {
		fmt.Fprintf(os.Stdout, `[]\n`)
		return
	}
	defer file.Close()

	reader := csv.NewReader(file)

	headers, _ := reader.Read([]string{})
	for range headers {
		continue
	}

	var persons []Person

	for {
		row, err := reader.Read()
		if err != nil {
			break
		}

		person := Person{}
		switch len(row) {
		case 4:
			person.Name = row[0]
			person.Age, _ = strconv.Atoi(row[1])
			person.Email = row[2]
			person.Score, _ = strconv.ParseFloat(row[3], 64)
		default:
			continue
		}
		if person.Name == "" {
			continue
		}
		persons = append(persons, person)
	}

	jsonBytes, err := json.Marshal(persons)
	if err != nil {
		os.Exit(1)
	}

	fmt.Println(string(jsonBytes))
}